### DISCUSSION TITLE
Add support for nested fields to with operator

### DISCUSSION THREAD

**Discussion by dgolub** [dotnet/csharplang#5766] [Language Ideas] [answered]
The new `with` operator is great for creating variations on immutable objects.  However, it gets a bit unwieldy when the object has nested objects that need to be modified.  It would be excellent if support could be added for dotted field names as syntactic sugar to simplify this case.  For example, if you wrote
```
x with {
    Foo.A = 1,
    Foo.B = 2
}
```
The compiler could translate it into
```
x with {
    Foo = x.Foo with {
        A = 1,
        B = 2
    }
}
```
To go a step further, if `Bar` is an array, then `x with { Bar[3] = 10 }` would copy over the array and replace the element at index 3 with 10.

**Comment by svick** [comment, marked as answer]
This is a subset of https://github.com/dotnet/csharplang/discussions/77.
